diff --git a/bin/ruby-build b/bin/ruby-build
index 70c1978..32fc0bd 100755
--- a/bin/ruby-build
+++ b/bin/ruby-build
@@ -1347,6 +1347,8 @@ fi
 PREFIX_PATH="${ARGUMENTS[1]}"
 if [ -z "$PREFIX_PATH" ]; then
   usage 1 >&2
+elif [ $(awk -F= '$1=="ID" { print $2; }' /etc/os-release) = "msys2" ]; then
+  PREFIX_PATH=$(cygpath -wa $PREFIX_PATH)
 elif [ "${PREFIX_PATH#/}" = "$PREFIX_PATH" ]; then
   PREFIX_PATH="${PWD}/${PREFIX_PATH}"
 fi
