diff --git a/bin/rbenv-install b/bin/rbenv-install
index 3d61f58..60fdb0b 100755
--- a/bin/rbenv-install
+++ b/bin/rbenv-install
@@ -257,6 +257,13 @@ fi
 # Execute `after_install` hooks.
 for hook in "${after_hooks[@]}"; do eval "$hook"; done
 
+# Remove trash from files without extension or *.cmd files (only for MSYS2).
+if [ $(awk -F= '$1=="ID" { print $2; }' /etc/os-release) = "msys2" ] && [ -d "${PREFIX}/bin" ]; then
+  for cmdfile in $(find "${PREFIX}/bin/*" ! -name "*.*" -o -name "*.cmd"); do
+    tail -n +9 "$cmdfile" > "${cmdfile}_" && mv "${cmdfile}_" "$cmdfile" && echo created file: "${cmdfile}"
+  done
+fi
+
 # Run `rbenv-rehash` after a successful installation.
 if [ "$STATUS" == "0" ]; then
   rbenv-rehash
