diff --git a/libexec/rbenv-rehash b/libexec/rbenv-rehash
index 84bd175..ce30b90 100755
--- a/libexec/rbenv-rehash
+++ b/libexec/rbenv-rehash
@@ -179,6 +179,13 @@ done
 install_registered_shims
 remove_stale_shims
 
+# Replace *.exe to *.bat files (only for MSYS2).
+if [ $(awk -F= '$1=="ID" { print $2; }' /etc/os-release) = "msys2" ] && [ -d "${RBENV_ROOT}/shims" ]; then
+  for exefile in $(find "${RBENV_ROOT}"/shims/* -name "*.exe"); do
+    mv "${exefile}" "${exefile%.exe}.bat"
+  done
+fi
+
 # Replace script in *.bat files (only for MSYS2).
 if [ $(awk -F= '$1=="ID" { print $2; }' /etc/os-release) = "msys2" ] && [ -d "${RBENV_ROOT}/shims" ]; then
   for batfile in $(find "${RBENV_ROOT}"/shims/* -name "*.bat"); do
